### PR TITLE
Fix unnecessary assigment false positive for out argument

### PR DESCRIPTION
**Pull request by maciek231**
Fix https://github.com/dotnet/roslyn/issues/58564

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs [hunk 1]
```diff
@@ -277,8 +277,21 @@ public static bool IsInLeftOfDeconstructionAssignment(this IOperation operation,
...
+        {
+            invocation = null;
+
+            if (operation.Parent is IArgumentOperation argumentOperation &&
+                    argumentOperation.Parameter?.RefKind == RefKind.Out)
+            {
+                invocation = argumentOperation.Parent as IInvocationOperation;
+            }
+
+            return invocation is not null;
```
Comment (DoctorKrolic) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:290]:
```suggestion
            if (operation.Parent is IArgumentOperation argumentOperation &&
                    argumentOperation.Parameter?.RefKind == RefKind.Out)
            {
               return argumentOperation.Parent is IInvocationOperation;
            }

            return false;
```

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs [hunk 2]
```diff
@@ -277,8 +277,21 @@ public static bool IsInLeftOfDeconstructionAssignment(this IOperation operation,
...
+        }
+
         /// <summary>
-        /// Retursn true if the given operation is a regular compound assignment,
+        /// Return true if the given operation is a regular compound assignment,
```
Comment (DoctorKrolic) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:294]: Return**s**)

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:290]: @DoctorKrolic `invocation` needs to be assigned as it's an `out` parameter. The suggestion doesn't assign it.

Comment (Youssef1313) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:290]:
```suggestion
            if (operation.Parent is IArgumentOperation { Parameter.RefKind: RefKind.Out, Parent: IInvocationOperation })
            {
                invocation = (IInvocationOperation)argumentOperation.Parent;
                return true;
            }

            invocation = null;
            return false;
```

Comment (maciek231) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:290]: @Youssef1313 It's assigned to null in line 282, or you mean something else?

Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Extensions/OperationExtensions.cs:290]: My comment was in a reply to another review comment that was deleted after I commented. You can ignore this comment :)

**Pull request closed without merging**